ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/V2/Args.hs

```diff
@@ -0,0 +1,32 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE TypeFamilies #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Ouroboros.Consensus.Storage.LedgerDB.V2.Args (
```

Comment:
I suggest that you add the LSM constructors, but do so with a `!Void` constructor argument. That way the code will have to explicitly branch where you've already written TODO comments about needing to check these arguments, but the "todo" branch can remain trivial for now. That seems better than keeping the TODO comments.

Reply:
A `!Void` makes it so that the cases result in redundant patterns:
```
src\ouroboros-consensus\Ouroboros\Consensus\Storage\LedgerDB\V2\Init.hs:214:8: error: [GHC-53633] [-Woverlapping-patterns, Werror=overlapping-patterns]
    Pattern match is redundant
    In a case alternative: LSMHandleArgs x -> ...
    |
214 |        LSMHandleArgs x    -> absurd x
    |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
What if you remove the bang?

Reply:
Done